### PR TITLE
Core integration

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Integrates parameters directly with openfisca-uk, adding a new `/parameters` endpoint that returns something like this:
```json
{
  "parameters": [
    {
      "default": 12570, 
      "description": "The basic rate is the first of three tax brackets on all income, after allowances are deducted.", 
      "summary": "Change the basic rate to @%", 
      "title": "basic_rate", 
      "type": "/1", 
      "value": 12570
    }, 
    {
      "default": 0.2, 
      "description": "The basic rate is the first of three tax brackets on all income, after allowances are deducted.", 
      "summary": "Change the basic rate to @%", 
      "title": "basic_rate", 
      "type": "/1", 
      "value": 0.2
    }
  ]
}
```

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
@MaxGhenis this also incorporates *some* of the new core server-side features, making `app.py` a bit cleaner. The above example now works locally, but I just need to transfer the metadata from the `default_policy.json` to the individual parameters in OpenFisca-UK.

**Comment by nikhilwoodruff** [comment]
Reason this test is failing is because of one household which, in 2022, if the personal allowance is increased by £200, loses around £200 in net income. This test passed in the openfisca-uk tests, because it only executed in 2021, but here the `use_current_parameters` reform backdates the mid-2021 values to the start of 2021.

**Comment by nikhilwoodruff** [comment]
OK, added parameter tests for all (but one, WA_adult_age), testing that they increase or decrease net income for all records in line with expectations (when increased at the margin).

File: policy_engine_uk/app.py [hunk 1]
```diff
@@ -1,199 +1,108 @@
...
-        mtr_chart=mtr,
-    )
-    if USE_CACHE:
-        blob.upload_from_string(json.dumps(result))
-    gc.collect()
-    duration = time() - start_time
-    app.logger.info(f"Situation reform completed ({round(duration, 2)}s)")
-    return result
-
+    def household_reform(self, params: dict = {}) -> dict:
```
Comment (MaxGhenis) [inline policy_engine_uk/app.py:72]: might be better practice to set to `None`: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

File: policy_engine_uk/simulation/reforms.py [hunk 2]
```diff
@@ -104,15 +86,68 @@ def formula(person, period, parameters):
...
+            title=meta["title"],
+            short_name=meta["short_name"],
+            description=meta["description"],
+            default=p(CURRENT_DATE),
+            value=p(CURRENT_DATE),
+            summary=meta["summary"],
+        )
+        default_values = dict(
+            min=0,
+            max=0,
```
Comment (MaxGhenis) [inline policy_engine_uk/simulation/reforms.py]: change to 1?

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**